test/ClangImporter/foreign_errors.swift

```diff
@@ -108,3 +113,24 @@ func testNSErrorExhaustive() {
     }
   }
 }
+
+func testBadOverrides(obj: FoolishErrorSub) throws {
+  try obj.performRiskyOperation()
+  let _: FoolishErrorSub = try obj.produceRiskyOutput()
+  let _: String = try obj.produceRiskyString()
+
+  let _: NSObject = try obj.badNullResult()
+  let _: CInt = try obj.badNullResult2() // This is unfortunate but consistent.
```

Comment:
@DougGregor, note this change. I checked the generated IR and there's not even a zero check here, because we treat "null" and "zero" as different error conventions (probably because of bridged types). But we don't crash, and the original code is arguably wrong. What do you think?